subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java

```diff
@@ -58,6 +60,19 @@ public void visitClassPath(URL[] classPath) {
         return implementationClassPath;
     }
 
+    public static Iterable<File> getClasspathFiles(ClassLoader classLoader) {
```

Comment:
might be better if `ClasspathUtil.getClasspath(ClassLoader)` returned a `ClassPath` instead, which then has a bunch of different query methods.